### ISSUE TITLE
model.py suggestions.

### ISSUE THREAD

**Issue by AustenLamacraft** [AustenLamacraft/audio-mps#6]
@mencia 
In the loss
~~~~tf.reduce_mean(loss) + self.h_reg * tf.reduce_sum(tf.square(self.H_diag)) + \
               self.r_reg * tf.real(tf.reduce_sum(tf.conj(self.R)*self.R))
~~~~
 Why `reduce_mean` rather than `reduce_sum`? You are choosing the relative weight of the main loss and the regularization. Taking the view that the whole thing is is a log probability suggests to me that it should be `reduce_mean`. What do you think?

Smaller things:

1. Lines 25-39. Are all the `if`s necessary? You supply the matrix as an initializer. If it's `None`, it does nothing.

1. `[self.rank_rho_0, self.bond_d]` is the same as `2*[self.rank_rho_0]`

1. Line 173. In `exp = tf.trace(tf.einsum('ab,cbd->cad', x, rho))` the `tf.trace` is redundant. Do it all with `einsum`.
